tempodb/encoding/vparquet3/block_traceql.go

```diff
@@ -833,6 +834,7 @@ var intrinsicColumnLookups = map[traceql.Intrinsic]struct {
 	traceql.IntrinsicTraceID:          {intrinsicScopeTrace, traceql.TypeString, columnPathTraceID},
 	traceql.IntrinsicTraceStartTime:   {intrinsicScopeTrace, traceql.TypeDuration, columnPathStartTimeUnixNano},
 
+	traceql.IntrinsicEventName:    {intrinsicScopeEvent, traceql.TypeNil, ""}, // Not used in vparquet3, this entry is only used to assign default scope.
```

Comment:
Since this isn't supported in vParquet2/3, I think we need to ensure that a query using event intrinsics doesn't execute.  Currently, it looks like the condition will be silently dropped while fetching, which means the query results could be incorrect:   the `{name=x && event:name=y}` will be executed against a vp2/3 block as `{name=x}`.    checkConditions could return ErrUnsupported if any event scope condition is present.

Reply:
Currently queries with `event:name` fail `categorizeConditions()`. But I agree, it's better to check this explicitly in `checkConditions()`. This also allows us to return a better error message